### PR TITLE
Refactor: using kube-pydantic instead of BaseResource

### PR DESCRIPTION
**Pull request by mperetzred**
Need to rebase on top of https://github.com/oadp-qe/openshift-adp-python-wrapper/pull/37

### CODE REVIEW

**Comment by mperetzred** [comment]
```bash
============================= test session starts ==============================
collecting ... collected 6 items

test_backup.py::test_custom_object_create 
test_backup.py::test_custom_object_merge 
test_backup.py::test_custom_object_replace 
test_backup.py::test_custom_object_delete 
test_backup.py::test_custom_object_get 
test_backup.py::test_custom_object_list 

============================== 6 passed in 1.45s ===============================

Process finished with exit code 0
PASSED                         [ 16%]PASSED                          [ 33%]PASSED                        [ 50%]PASSED                         [ 66%]PASSED                            [ 83%]PASSED                           [100%]
```
all mocked tests have passed

File: openshift_resources/drivers/apicore.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from ocp_resources.resource import _get_api_version
+
+
+class NewK8SCustomApiCore(K8SCustomApiCore):
+
+    def __init__(self, *args, model=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        if kwargs.get("config") and not isinstance(kwargs.get("config"), MagicMock):
+            self.dyn_client = DynamicClient(self.api.client)
+            self.version = _get_api_version(self.dyn_client, self.group, model)
```
Comment (amastbau) [inline openshift_resources/drivers/apicore.py:14]: oh ok

Comment (amastbau) [reply openshift_resources/drivers/apicore.py:14]: maybe move this one to our infra
(and the other statics)

File: openshift_resources/drivers/apicore.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from unittest.mock import MagicMock
+
+from kubernetes.dynamic import DynamicClient
+from kubernetes_pydantic.drivers.apicore import K8SCustomApiCore
+from ocp_resources.resource import _get_api_version
+
+
+class NewK8SCustomApiCore(K8SCustomApiCore):
```
[the remark this answers is not in the document]
Comment (mperetzred) [reply openshift_resources/drivers/apicore.py:8]: yes, I just wanted to add this dynamic API version discovery, so I overloaded the class. Can't think of a better name, please feel free to suggest.

**Comment by mperetzred** [comment]
depends on https://github.com/oadp-qe/openshift-adp-python-wrapper/pull/39/files

File: openshift_resources/drivers/apicore.py (hunk 1 above)
Comment (mperetzred) [reply openshift_resources/drivers/apicore.py:14]: which infra?

**Review by amastbau — APPROVED** [review]

**Pull request merged**